### DISCUSSION TITLE
Rearrange and Hide Spells

### DISCUSSION THREAD

**Discussion by Cinemacation** [runelite/runelite#16558] [Ideas]
Not sure if rearranging spells in a spellbook is possible in the client, but thought I might as well suggest it incase it is.

**How I See It Working:**
Have an example spellbook in the plugin settings:
- Users can drag spells to arrange them how they want
- Clicking on a spell will change it's status from Visible to Hidden
- Hidden spells will act a placeholder, so spells won't shift down
- 4 Tabs; 1 for each spellbook
- "Revert to default" Button incase the user is confused

Possible Extra:
- "Save Layout" & "Load Layout" Options

![image](https://user-images.githubusercontent.com/64283901/236600419-ebf68724-9d7f-40c5-a897-4939bdeceae0.png)

**Comment by Nightfirecat** [comment]
We're not allowed to do this. From Jagex's [Third Party Client Guidelines](https://secure.runescape.com/m=news/third-party-client-guidelines?oldschool=1):

> Can unhide, hide, move, or resize interface components on the Prayer and Spellbook interfaces

**Reply by Cinemacation** [reply]
Dang it, thanks for the quick response though. Appreciate it.
